### ISSUE TITLE
Run Xamarin.UITest for iOS locally on Windows via connected Mac's iOS simulator

### ISSUE THREAD

**Issue by mzhukovs** [microsoft/appcenter#574]
**Describe the solution you'd like**
Currently, for Visual Studio **Windows** users, we are not able to run Xamarin.UITest on an iOS Simulator via the connected Mac. It would be great to be able to do so, since we can already run the app on iOS Simulator from VS Windows, and even remote the simulator back to VS on Windows.

**Describe alternatives you've considered**
In this case, the only available alternative is to use VS for Mac, which is a non-trivial hassle if you're developing on VS Windows.

**Comment by Oddj0b** [comment]
Thank you for your feedback, I'll have to investigate a little before we can proceed with this.

**Comment by mzhukovs** [comment]
Thanks @Oddj0b  - the obvious upside is that it would be great and a lot more convenient to be able to verify the tests actually work on iOS as intended first before submitting them to App Center.

**Comment by mzhukovs** [comment]
Just checking in - any thoughts, at least initially? @Oddj0b    please let me know when you get a chance, thanks!

**Comment by Oddj0b** [comment]
@mzhukovs I like the idea, but it will be hard to achieve. If we can get more reactions it's more likely to be picked-up.

**Comment by curia-damiano** [comment]
Hi Villars, I would also like to give my support to this proposal!

**Comment by mzhukovs** [comment]
@Oddj0b well then it's times like these that I wish I had a blog/twitter account with a mass following of other like-minded geeks.

**Comment by scott225** [comment]
I'd also appreciate this feature. I'm creating a testing framework that will tie in the ability to test websites, databases, UWP, Android and iOS apps. The framework will run on a windows server and needs to be able to automate testing using a remote mac server. If the code needs to be executed on a mac, the entire framework will need to exist on a mac server. From someone who is lifelong Windows user, this reliance on a mac server would be expensive in terms of educating/ramping myself up and the actual cost of a mac server.

**Comment by gpremkamal** [comment]
I do support this feature.

**Comment by maurjiu** [comment]
+1

**Comment by andrewmorris-dev** [comment]
+1

**Comment by mzhukovs** [comment]
@Oddj0b things are heating up!

**Comment by GeorgeDarakchiev** [comment]
+1

**Comment by sspasova** [comment]
+1

**Comment by didiyordanova** [comment]
+1

**Comment by rambo406** [comment]
+1

**Comment by andresdsep** [comment]
+1

**Comment by cesar1414** [comment]
hi all 
I have this same problem 
someone has an advice ?

**Comment by curia-damiano** [comment]
Hi Cesar, please support us requesting the implementation of this feature

**Comment by DanielGlick45** [comment]
+1

**Comment by Brosten** [comment]
+1 from me!
About to start UI testing out Android/iOS Xam-forms app. 100% Development on Windows. Setting all up to connect to TFVC (yes, we are moving to git, but not yet) is a mess on Mac. Any status or timeplan for this feature would be great!

**Comment by mzhukovs** [comment]
@Oddj0b I hear soon that we'll be able to plug an apple device directly into our Windows machine for debugging, this feature request seems like it would align nicely with that. What are your thoughts on this having a chance at making it into the roadmap?

**Comment by Oddj0b** [comment]
@mzhukovs That sounds very interesting, do you have a source on it? Because any chance that we can make it easier, I'll try and grab it.

**Comment by Brosten** [comment]
What about the remote simulator? Debugging from VS on windows, getting a remote simulator on your Windows screen is working today. It isn't possible to use that method to run tests, at lest on a iOS simulator?

**Comment by Oddj0b** [comment]
@Brosten Good question, I'll have to investigate that a little more.  Do you have a good guide handy?

**Comment by Brosten** [comment]
@Oddj0b 
https://docs.microsoft.com/en-us/xamarin/tools/ios-simulator/

**Comment by Oddj0b** [comment]
@Brosten thank you for the link, I'll include it in my investigations.

**Comment by mzhukovs** [comment]
@Oddj0b https://channel9.msdn.com/Shows/XamarinShow/Debug-to-iOS-Devices-Over-Wi-Fi

**Comment by mastermoe01** [comment]
+1

**Comment by Dresel** [comment]
Any updates on this?

**Comment by ilkeSylmz** [comment]
+1

**Comment by jmichas** [comment]
+1

**Comment by pooter07** [comment]
+1

**Comment by CrustyApplesniffer** [comment]
👍

**Comment by granth7** [comment]
+1

**Comment by Csaba8472** [comment]
+1

**Comment by evgenyvalavin** [comment]
+1

**Comment by VahidShir** [comment]
+1

**Comment by curia-damiano** [comment]
+1 a second time because Android 11 is coming and we do need it

**Comment by reggiebIV** [comment]
I, too, would like to add another +1.

**Comment by RobertBickers** [comment]
+1 👍

**Comment by smardine** [comment]
It would be awsome +1

**Comment by xamadev** [comment]
+1

**Comment by mzhukovs** [comment]
@Oddj0b hi Villars, still no concrete plans for this? Saw that you actually removed your assignment above.

**Comment by stefanhk31** [comment]
+1. Xamarin UITests are currently unusable for my team because of this issue.

**Comment by DerekFoulk** [comment]
+1000 This should've been implemented from the start (in my humble opinion). I'm sure it won't get picked up since Xamarin is going to be phased out in the near future- but MAUI must support UI testing of iOS applications from our Windows workstations...

**Comment by mzhukovs** [comment]
@davidortinau will running tests for iOS from a Windows machine be possible in MAUI as per current plans?

**Comment by LukasBajus** [comment]
+1
Is there any progress on this? I only can run my tests on Android simulators now. I need to run them also on iOS emulator to find out what needs to be changed in tests so they run correctly also on iOS.

**Comment by mikkelAtVisma** [comment]
+1

**Comment by LukasBajus** [comment]
Hi. Is there any plan for this. It is really blocking me with creating tests for iOS. 
Thank you.

**Comment by Jeroen5356** [comment]
+1

**Comment by LukasBajus** [comment]
Hey. Still no status on this? Can somebody please have a look on this?

**Comment by Welchen** [comment]
+1

**Comment by Aresak** [comment]
+1

**Comment by LukasBajus** [comment]
Still no work on this? Is there at least some workaround or something to run my code on my Windows workstation on iPhone emulator?

**Comment by LeoJHarris** [comment]
Jumping on board this ticket, I too would love to see this work in windows machine 👍

**Comment by andrewnemecek** [comment]
This would be helpful.

**Comment by Mahesh5R** [comment]
It will be really great....👍.....+1

**Comment by smit1712** [comment]
This would make my life so much easier

**Comment by LukasBajus** [comment]
OK I'll ask in another way. Is it somehow possible to see how my iPhone tests works (debug them) on Windows machine?

**Comment by curia-damiano** [comment]
> OK I'll ask in another way. Is it somehow possible to see how my iPhone tests works (debug them) on Windows machine?

Sadly not that I am aware... I am awaiting for this feature since two years and I won't give up.

Btw there are other features to bring from VS4Mac to VS4Win:
- reference Test Project in solution (currently not possible, you need to add reference to .apk/.ipa)
- reintroduce Unit Test template projects (available in the past, removed from VS4Win)

**Comment by smardine** [comment]
Do not close it please it's a very nice feature and lot of people want it 😁

**Comment by DerekFoulk** [comment]
Yes, please don't close this. This has been a big hole in our workflow.

**Comment by LukasBajus** [comment]
how can you close it just like that? can somebody have a look on it? it has been ignored for a long time. Of course there is no activity on it, because I was already tired of writing here and talking to myself...

**Comment by Brosten** [comment]
Unfortunately it seems like all work with AppCenter is halted long time ago. I guess MS have some plan they havn't yet made public. Moving things to GitHub? Closing AppCenter or whatever, but I wouldn't hold my breath waiting for things to progress in AppCenter.

**Comment by MichaelSafro** [comment]
Joining the plea not to close the issue, but provide a solution instead!

**Comment by curia-damiano** [comment]
+1 please don't close it

**Comment by ricardomomm** [comment]
+1

**Comment by LukasBajus** [comment]
again. please don't close it. This still doesn't work, and no workaround or solution has been provided.

**Comment by marcos-marimon** [comment]
I do support this feature, too.

**Comment by martinthesmith** [comment]
+1 It shouldn't be closed.

**Comment by mattwood2855** [comment]
Its time to add this, MAUI wont wow people enough if you cant fully test.

**Comment by andrea-gavioli** [comment]
+1 UI tests are extremely important and should be super nice to have this feature

**Comment by Don-Antonio** [comment]
Even though we're in 2023, this feature remains unsupported.
Disappointed

**Comment by jmichas** [comment]
Have any of you tried to use Appium to accomplish this? 

I switched to developing on Mac with VS Mac and switched over all my Xam UI Tests to Appium using SpecFlow, it works much better than Xam UI used to, I always had problems running my tests reliably.  Now my Appium code spins up docker instances and runs tests in simulator/emulator or physical attached devices and spins my test docker stuff down when done.

Might be worth checking out if you can trigger UI tests from windows using Appium.

**Comment by LukasBajus** [comment]
@jmichas 
Thanks for the tip. One follow-up question, is your application written in xamarin.forms ? Cause at the beginning I was trying to use Appium with our app, but it was not working very well as our app is in xamarin.forms . Maybe something changed in those three years and it actually works now.

**Comment by jmichas** [comment]
@LukasBajus Yeah, ours is a xam forms app too. There was a bit of a learning curve and I wrote some extension methods to minimize the amount of xam UI test code I needed to refactor, but once I got it working I feel it is more powerful than xam ui and less problematic. One problem I always ad with xam ui was with the calabash agent not running or connecting. Appium is much more stable for me. But I also have not tackled running the UI tests as part of CI/CD build, I run my tests manually offline as a last QA before merging major code updates.
